### PR TITLE
Optimize detail page

### PR DESCRIPTION
**Pull request by HuiminFrontend**

### CODE REVIEW

File: src/pages/Details.js [hunk 1]
```diff
@@ -9,6 +10,17 @@ import Navigation from '../components/Navigation/Navigation';
...
+  let actives = [false, false, false, false, false];
+
+  function setActive() {
+    for (let i = 0; i < currentMenu.spiciness; i++) {
+      if (currentMenu.spiciness < actives.length) {
+        actives[i] = true;
+      }
+    }
+  }
+  setActive();
```
Comment (PReichetanz) [inline src/pages/Details.js]: I think "active" is a too general name and setActive is reserved for state setters. Additionally, I would integrate the array into the function, like so:

```suggestion
 let activePeppers = calculateActivePeppers();

  function calculateActivePeppers() {
    let pepperStatus = [false, false, false, false, false];
      for (let i = 0; i < currentMenu.spiciness; i++) {
        if (currentMenu.spiciness < pepperStatus.length) {
          pepperStatus[i] = true;
        }
      }
      return pepperStatus;
    }
```

File: src/pages/Details.js [hunk 2]
```diff
@@ -23,20 +35,25 @@ export default function Details({menusdata}) {
...
+                <H2>Taste: </H2>
                 {currentMenu.taste}
               </Section>
               <Section>
-                <h2>Spiciness:</h2>
-                {currentMenu.spiciness} / 5
+                <H2>Spiciness:</H2>
+
+                <div spiciness={currentMenu.spiciness}>
+                  {actives.map((active, index) => {
```
Comment (PReichetanz) [inline src/pages/Details.js]:
```suggestion
                  {activePeppers.map((active, index) => {
```

**Review by PReichetanz — COMMENTED** [review]
Looks good to me, just an idea how to further improve your calculation of the red peppers ;)

File: src/pages/Details.js (hunk 1 above)
Comment (stefankorder) [inline src/pages/Details.js]: You can use this function:

```
function calculatePeppers(spicinessValue) {
  return [false, false, false, false, false].map(
    (_, index) => spicinessValue > index
  );
}
```

File: src/pages/Details.js (hunk 2 above)
Comment (stefankorder) [inline src/pages/Details.js]: Now here you can call the writen function and map over the returned array.

```suggestion
                  {calculatePeppers(spicinessValue).map((active, index) => {
```

**Review by stefankorder — CHANGES_REQUESTED** [review]
Good work! Take a look at my suggestions 😃

File: src/pages/Details.js (hunk 1 above)
Comment (HuiminFrontend) [reply src/pages/Details.js]: Thank you, I never seen this before, looks cool!  I'll go find more info about it.

**Pull request merged**